network/src/protocols/wire/handshake/v1/mod.rs

```diff
@@ -14,10 +14,15 @@ use libra_config::network_id::NetworkId;
 use libra_types::chain_id::ChainId;
 use serde::{export::Formatter, Deserialize, Serialize};
 use std::{collections::BTreeMap, convert::TryInto, fmt, iter::Iterator};
+use thiserror::Error;
 
 #[cfg(test)]
 mod test;
 
+//
+// ProtocolId
+//
+
```

Comment:
I never understood the purpose of these types of comments. As a vim developer I search for enum ProtocolId ... if we need these, we should just separate them into different files.

Reply:
It's not for the actual struct or enum or whatever, but to group related structs, functions, implementations, etc. together. Before this refactor ProtocolId implementations where scattered all around the file and it was quite the challenge to read it : o right now it should be smoother for first-time comer or even for us to find things. The comments are really here to separate these things and force you to place new implementations where they belong :)

Reply:
I agree, cut the comments. We should stick to rust doc comments and not add these sectional comments.

Personally, if the file is so big that you need these "sectional" comments, there should be separate files for these components.  However, I don't want you to move the enums and impl right now.

Reply:
I don't agree with that, you don't split a file that is disorganized, you organize it. If it's too large sure, but that's not the problem here. This is what other crates like crypto do as well, and it greatly helps readability. It'd be good if we encouraged PRs that improve readability imo, instead of sticking to a block of code that someone needs to reverse engineer to understand. I think long-term the handshake can be simplified and optimized and this is a step in the right direction